compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -147,8 +148,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 if src.layout.size != dest.layout.size {
                     let src_bytes = src.layout.size.bytes();
                     let dest_bytes = dest.layout.size.bytes();
-                    let src_ty = format!("{}", src.layout.ty);
-                    let dest_ty = format!("{}", dest.layout.ty);
+                    let src_ty = with_no_trimmed_paths!(format!("{}", src.layout.ty));
+                    let dest_ty = with_no_trimmed_paths!(format!("{}", dest.layout.ty));
```

Comment:
throw_ub_custom should be able to take anything that's `IntoDiagnosticArg` -- does this work if you just do:

```suggestion
                    let src_ty = src.layout.ty;
                    let dest_ty = dest.layout.ty;
```